public/app/features/dashboard-scene/scene/PanelNonApplicableDrilldownsSubHeader.tsx

```diff
@@ -58,17 +58,19 @@ export function PanelNonApplicableDrilldownsSubHeader({ filtersVar, groupByVar,
       const applicability = await getDrilldownApplicability(queryRunner, filtersVar, groupByVar);
       if (filterValues.length) {
         const nonApplicableFilters = filterValues.filter((filter) => {
-          const result = applicability?.find((entry) => entry.key === filter.key && entry.origin === filter.origin);
-          return !result?.applicable;
+          const result = applicability?.find(
+            (entry) => entry.key === filter.key && entry.origin === filter.origin && entry.type === 'filter'
+          );
+          return result && !result.applicable;
```

Comment:
fixed checks here as it returned wrongly, also added an extra property to make sure we differentiate between same filter and groupBy keys